src/base/QXmppMixInvitation.cpp

```diff
@@ -0,0 +1,176 @@
+{
+    d->inviterJid = inviterJid;
+}
+
+///
+/// Returns the JID of the invitee.
+///
+/// \return the invitee's JID
+///
+QString QXmppMixInvitation::inviteeJid() const
+{
+    return d->inviteeJid;
+}
+
+///
+/// Sets the JID of the invitee.
+///
+/// \param inviteeJid invitee's JID
+///
+void QXmppMixInvitation::setInviteeJid(const QString &inviteeJid)
+{
+    d->inviteeJid = inviteeJid;
+}
+
+///
+/// Returns the JID of the channel.
+///
+/// \return the channel's JID
+///
+QString QXmppMixInvitation::channelJid() const
+{
+    return d->channelJid;
+}
+
+///
+/// Sets the JID of the channel.
+///
+/// \param channelJid channel JID
+///
+void QXmppMixInvitation::setChannelJid(const QString &channelJid)
+{
+    d->channelJid = channelJid;
+}
+
+///
+/// Returns the token which is generated by the server and used by the invitee for authentication.
+///
+/// \return the generated token used for authentication
+///
+QString QXmppMixInvitation::token() const
+{
+    return d->token;
+}
+
+///
+/// Sets the token which is generated by the server and used by the invitee for authentication.
+///
+/// \param token authentication token
+///
+void QXmppMixInvitation::setToken(const QString &token)
+{
+    d->token = token;
+}
+
+///
+/// Determines whether the invitation object is initialized.
+///
+/// \return true if all of the object's attributes are empty, otherwise false
+///
+bool QXmppMixInvitation::isNull() const
+{
+    return d->inviterJid.isEmpty() && d->inviteeJid.isEmpty() && d->channelJid.isEmpty() && d->token.isEmpty();
+}
+
+void QXmppMixInvitation::parse(const QDomElement &element)
+{
+    d->inviterJid = element.firstChildElement(QStringLiteral("inviter")).text();
+    d->inviteeJid = element.firstChildElement(QStringLiteral("invitee")).text();
+    d->channelJid = element.firstChildElement(QStringLiteral("channel")).text();
+    d->token = element.firstChildElement(QStringLiteral("token")).text();
+}
+
+void QXmppMixInvitation::toXml(QXmlStreamWriter* writer) const
+{
+    if (isNull())
+        return;
+
+    writer->writeStartElement(QStringLiteral("invitation"));
+    writer->writeDefaultNamespace(ns_mix_misc);
+
+    helperToXmlAddTextElement(writer, QStringLiteral("inviter"), d->inviterJid);
+    helperToXmlAddTextElement(writer, QStringLiteral("invitee"), d->inviteeJid);
+    helperToXmlAddTextElement(writer, QStringLiteral("channel"), d->channelJid);
+    helperToXmlAddTextElement(writer, QStringLiteral("token"), d->token);
+
+    writer->writeEndElement();
+}
+
+///
+/// Determines whether the given DOM element is a MIX invitation.
+///
+/// \param element DOM element being checked
+///
+/// \return true if element is a MIX invitation, otherwise false
+///
+bool QXmppMixInvitation::isMixInvitation(const QDomElement &element)
+{
+    return element.tagName() == QStringLiteral("invitation") && (element.namespaceURI().isEmpty() || element.namespaceURI() == ns_mix_misc);
```

Comment:
In which case should we accept an empty namespace? As namespaces are inherited, the namespace is never going to be empty in an XMPP stream.